fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -1654,25 +1654,32 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 		}
 		if (SERVER_KNOBS->ROCKSDB_HISTOGRAMS_SAMPLE_RATE > 0) {
 			collection = actorCollection(addActor.getFuture());
-			addActor.send(updateHistogram(metricPromiseStream.getFuture()));
+		}
+		for (int i = 0; i < SERVER_KNOBS->ROCKSDB_READ_PARALLELISM + 1; i++) {
+			// ROCKSDB_READ_PARALLELISM readers and 1 writer
+			metricPromiseStreams.emplace_back(
+			    std::make_unique<ThreadReturnPromiseStream<std::pair<std::string, double>>>());
+			if (SERVER_KNOBS->ROCKSDB_HISTOGRAMS_SAMPLE_RATE > 0) {
+				addActor.send(updateHistogram(metricPromiseStreams[i]->getFuture()));
+			}
+		}
+		if (SERVER_KNOBS->ROCKSDB_HISTOGRAMS_SAMPLE_RATE > 0) {
+			actorErrorListener = errorListenActor(collection);
 		}
 		writeThread->addThread(new Writer(db,
 		                                  defaultFdbCF,
 		                                  id,
 		                                  readIterPool,
 		                                  perfContextMetrics,
 		                                  SERVER_KNOBS->ROCKSDB_READ_PARALLELISM,
-		                                  &metricPromiseStream),
+		                                  *(metricPromiseStreams[SERVER_KNOBS->ROCKSDB_READ_PARALLELISM])),
```

Comment:
It's not recommended to pass non-const object by reference. To solve the actor exception due to early destruction of the promise stream, we could either pass the promise stream by pointer or cancel the actors before the threads are [stopped](https://github.com/apple/foundationdb/blob/327c1da0a0dc456f3cf4b808ce27d1e6eda57f31/fdbserver/KeyValueStoreRocksDB.actor.cpp#L1754). The later is more reasonable. However, feel free to use a pointer if the later one is adding unnecessary complexity.